Core/include/Acts/Geometry/CylinderContainerBlueprintNode.hpp

```diff
@@ -0,0 +1,151 @@
+  ///   ` connect` and collect the returned shell
+  /// -# Produce a combined @ref Acts::CylinderStackPortalShell from all the shells
+  /// -# Return that shell representation
+  ///
+  /// @param options The global blueprint options
+  /// @param gctx The geometry context (nominal usually)
+  /// @param logger The logger to use
+  /// @return The combined @ref Acts::CylinderStackPortalShell
+  CylinderStackPortalShell& connect(
+      const BlueprintOptions& options, const GeometryContext& gctx,
+      const Logger& logger = Acts::getDummyLogger()) override;
+
+  /// This participates in the construction of the geometry via the blueprint
+  /// tree. The steps are approximately as follows:
+  /// -# Register portals created for gap volumes, as they're not handled by
+  ///    dedicated nodes
+  /// -# Register gap volumes in the @p parent volume
+  /// -# Create a configured @ref Acts::INavigationPolicy for the gap
+  /// -# Call `finalize` on all children while passing through @p parent.
+  ///
+  /// @param options The global blueprint options
+  /// @param gctx The geometry context (nominal usually)
+  /// @param parent The parent volume
+  /// @param logger The logger to use
+  void finalize(const BlueprintOptions& options, const GeometryContext& gctx,
+                TrackingVolume& parent, const Logger& logger) override;
+
+  /// Setter for the stacking direction
+  /// @param direction The stacking direction
+  /// @return This node for chaining
+  CylinderContainerBlueprintNode& setDirection(BinningValue direction);
+
+  /// Setter for the attachment strategy
+  /// @param attachmentStrategy The attachment strategy
+  /// @return This node for chaining
+  CylinderContainerBlueprintNode& setAttachmentStrategy(
+      CylinderVolumeStack::AttachmentStrategy attachmentStrategy);
+
+  /// Setter for the resize strategy
+  /// @param resizeStrategy The resize strategy
+  /// @return This node for chaining
+  CylinderContainerBlueprintNode& setResizeStrategy(
+      CylinderVolumeStack::ResizeStrategy resizeStrategy);
+
+  /// Accessor to the stacking direction
+  /// @return The stacking direction
+  BinningValue direction() const;
+
+  /// Accessor to the attachment strategy
+  /// @return The attachment strategy
+  CylinderVolumeStack::AttachmentStrategy attachmentStrategy() const;
+
+  /// Accessor to the resize strategy
+  /// @return The resize strategy
+  CylinderVolumeStack::ResizeStrategy resizeStrategy() const;
+
+ private:
+  /// @copydoc BlueprintNode::addToGraphviz
+  void addToGraphviz(std::ostream& os) const override;
+
+  /// Helper function to check if a volume was created as a gap volume.
+  /// @param volume The volume to check
+  /// @return True if the volume is a gap volume, false otherwise
+  bool isGapVolume(const Volume& volume) const;
+
+  std::string m_name;
+
+  BinningValue m_direction;
+
+  CylinderVolumeStack::AttachmentStrategy m_attachmentStrategy;
+
+  CylinderVolumeStack::ResizeStrategy m_resizeStrategy;
```

Comment:
potentially uninitialized?

Reply:
It's defaulted in the constructor, but I'll add defaults here to be safe.

Reply:
Oh I see - I guess we can only pick a random one and comment that this is to be overwritten by the user

Reply:
It'll be overwritten by the constructor in any case